modules/rostests/apitests/version/VerQueryValue.c

```diff
@@ -0,0 +1,293 @@
+    ok(result, "GetFileVersionInfoA failed\n");
+
+    /* Query available translations */
+    result = VerQueryValueA(pvVersionInfoA,
+                            "\\VarFileInfo\\Translation",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 4, "Unexpected value\n");
+    ok((cbLen & 0x3) == 0, "Unexpected value\n");
+    uLanguage = ((USHORT*)pvData)[0];
+    uCodePage = ((USHORT*)pvData)[1];
+
+    /* Query sublock */
+    sprintf(szSubBlock, "\\StringFileInfo\\%04X%04X\\CompanyName", uLanguage, uCodePage);
+    result = VerQueryValueA(pvVersionInfoA,
+                            szSubBlock,
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 2, "Unexpected value\n");
+    ok((cbLen & 0x1) == 0, "Unexpected value\n");
+
+    free(pvVersionInfoA);
+}
+
+void
+Test_VerQueryValueW(void)
+{
+    DWORD dwVersionInfoSizeW;
+    PVOID pvVersionInfoW, pvData = NULL;
+    WCHAR szSubBlock[256];
+    USHORT uLanguage, uCodePage;
+    UINT cbLen;
+    BOOL result;
+
+    /* Get the size for the unicode version info */
+    dwVersionInfoSizeW = GetFileVersionInfoSizeW(L"kernel32.dll", NULL);
+    ok(dwVersionInfoSizeW > 0, "GetFileVersionInfoSizeW failed\n");
+
+    /* Get the unicode version info from kernel32 */
+    pvVersionInfoW = malloc(dwVersionInfoSizeW);
+    result = GetFileVersionInfoW(L"kernel32.dll", 0, dwVersionInfoSizeW, pvVersionInfoW);
+    ok(result, "GetFileVersionInfoW failed\n");
+
+    /* Query available translations */
+    result = VerQueryValueW(pvVersionInfoW,
+                            L"\\VarFileInfo\\Translation",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 4, "Unexpected value\n");
+    ok((cbLen & 0x3) == 0, "Unexpected value\n");
+    uLanguage = ((USHORT*)pvData)[0];
+    uCodePage = ((USHORT*)pvData)[1];
+
+    /* Query sublock */
+    _swprintf(szSubBlock, L"\\StringFileInfo\\%04X%04X\\CompanyName", uLanguage, uCodePage);
+    result = VerQueryValueW(pvVersionInfoW,
+                            szSubBlock,
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueA failed\n");
+    ok(cbLen >= 2, "Unexpected value\n");
+    ok((cbLen & 0x1) == 0, "Unexpected value\n");
+
+    free(pvVersionInfoW);
+}
+
+void
+Test_StaticVersionnfo(void)
+{
+    PVOID pvData = NULL;
+    SIZE_T ExpectedOffset;
+    UINT cbLen;
+    BOOL result;
+
+    /* Test a special static version */
+    result = VerQueryValueW(&g_VersionInfo,
+                            L"\\StringFileInfo\\000004B0\\CompanyName",
+                            &pvData,
+                            &cbLen);
+    ok(result, "VerQueryValueW failed\n");
+    ok_eq_int(cbLen, 16);
+    ok(wcscmp((WCHAR*)pvData, L"ReactOS Project") == 0, "Unexpected value\n");
```

Comment:
in `apitest.h` we defined `ok_eq_wstr` that might be of use here:
`#define ok_eq_wstr(value, expected)         ok(!wcscmp(value, expected), #value " = \"%ls\", expected \"%ls\"\n", value, expected)`